src/delay.ml

```diff
@@ -307,6 +307,74 @@ let parse_durationl l =
   let s = String.concat " " l in
   parse_duration s
 
+let duration_of_daypack_duration (x : Daypack_lib.Duration.t) : duration =
+  C.Period.lmake ~day:x.days ~hour:x.hours ~minute:x.minutes ~second:x.seconds ()
+
+let get_current_tz_offset_s () =
+  Ptime_clock.current_tz_offset_s () |> Option.get |> Option.some
+
+let search_param =
+  let open Daypack_lib in
+  Search_param.Years_ahead_start_unix_second {
+    search_using_tz_offset_s = get_current_tz_offset_s ();
+    start = Time.Current.cur_unix_second ();
+    search_years_ahead = 5;
+  }
+
+let parse_durationl' l =
+  let s = String.concat " " l in
+  Daypack_lib.Duration.of_string s
+  |> Result.map duration_of_daypack_duration
+
+let parse_time_pointl' l =
+  let open Daypack_lib in
+  let s = String.concat " " l in
+  match Time_expr.of_string s with
+  | Ok x ->
+    (match Time_expr.next_match_time_slot search_param
+             x with
+    | Error msg -> Error msg
+    | Ok x ->
+      match x with
+      | None -> Error "Failed to find a matching time point"
+      | Some (time_slot_start, _time_slot_end) ->
+        let cur = Time.Current.cur_unix_second () in
+        let diff = if time_slot_start >= cur then Int64.sub time_slot_start cur else 0L in
+        Duration.of_seconds diff
+        |> Result.get_ok
+        |> duration_of_daypack_duration
+        |> Result.ok
```

Comment:
Are you fine with replacing type `Delay.duration` with `Daypack_lib.Duration.t`? I was trying to address this, but can't seem to do it without duplication.

Reply:
I wouldn't mind, but in that case, we might as well replace all uses of calendar.

Have you tested your implementation against calendar ? :3

Reply:
> I wouldn't mind, but in that case, we might as well replace all uses of calendar.

Comparing feature parity of daypack to calendar actually never crossed my mind, since it was calendar before the switch to ptime...good reminder...

> Have you tested your implementation against calendar ? :3

(Sweating intensifies)

I am not confident in duration in daypack working as robustly as its counterpart in calendar (period) since there aren't tests on duration yet...(opening an issue)